src/DocumentDB.ChangeFeedProcessor/PartitionManagement/PartitionSupervisor.cs

```diff
@@ -63,11 +63,16 @@ public async Task RunAsync(CancellationToken shutdownToken)
             {
                 closeReason = ChangeFeedObserverCloseReason.Shutdown;
             }
-            catch (Exception) when (processorTask.IsFaulted)
+            catch (Exception ex) when (processorTask.IsFaulted && ex is ObserverException)
             {
                 closeReason = ChangeFeedObserverCloseReason.ObserverError;
                 throw;
             }
+            catch (Exception) when (processorTask.IsFaulted)
```

Comment:
It's arguable whether this is needed or not (Unknown/Shutdown is by default and this would override Shutdown), but OK to keep.